lib/notifications/display.dart

```diff
@@ -139,6 +142,22 @@ class NotificationDisplayManager {
         // TODO this doesn't set the Intent flags we set in zulip-mobile; is that OK?
         //   (This is a legacy of `flutter_local_notifications`.)
         ),
+      autoCancel: true,
+    );
+
+    await ZulipBinding.instance.androidNotificationHost.notify(
+      id: notificationIdAsHashOf(groupKey),
+      channelId: NotificationChannelManager.kChannelId,
+      tag: groupKey,
+      groupKey: groupKey,
+      isGroupSummary: true,
+      color: kZulipBrandColor.value,
+      // TODO vary notification icon for debug
+      smallIconResourceName: 'zulip_notification', // This name must appear in keep.xml too: https://github.com/zulip/zulip-flutter/issues/528
+      inboxStyle: InboxStyle(
+        // TODO(#570) Show organization name, not URL
+        summaryText: data.realmUri.toString()),
+      autoCancel: true,
```

Comment:
There's seems to be no difference in behavior with it added or not, things verified in both conditions:
- Tapping on summary header, which does nothing; unless clicked on the collapse icon — which toggles between collapsed state.
- Tapping on child notif, opens the app and routes to the correct message, also clears that specific notif from the panel/statusbar (only if child notif had `setAutoCancel(true)`).
- Keep opening (which also clears them) child notifs, and made sure summary notif automatically dismisses when there are no child notifs anymore.

Reply:
From [Stackoverflow](https://stackoverflow.com/questions/45590860/group-summary-notification-not-canceled-when-last-containing-notification-is-can#comment107044464_53043333), it seems like the summary notification may linger if child notifs are removed programmatically and `setAutoCancel(true)` on summary notif is not set. So, it may be a problem when #341 is implemented, but this is just a speculation though, will have to test and see there's really a difference while implementing #341.

Reply:
Interesting, thanks for the investigation.

That issue where the summary lingers reminds me of a bug we briefly had in zulip-mobile:
https://github.com/zulip/zulip-mobile/issues/5119
Looks like that didn't involve a lack of `setAutoCancel(true)`, though ­— we had that on the summary notification from the beginning (in zulip/zulip-mobile@4229325b4e3421b4c04c6939907ba19945151521).

Given that this logically shouldn't do anything (based on the reasoning in that TODO comment), and that we've also tested empirically and can't find it doing anything, let's leave it out. We can always add it back if we later discover after #341 that it's needed as a hack.

In any case, we definitely should not include the line while leaving out the TODO comment that was attached to it. That comment is useful information, which we shouldn't discard.